source/agora/common/crypto/Schnorr.d

```diff
@@ -374,6 +399,66 @@ nothrow @nogc @safe unittest
     assert(stolen_key != kp.v);
 }
 
+/// Single-signer as part of multi-sig
+public Signature multiSigSign (const ref Pair R,
+    const ref Scalar v, const ref Scalar c) nothrow @nogc @safe
+{
+    Scalar s = R.v + (v * c);
+    return Sig(R.V, s).toBlob();
+}
+
+/// multi-sig verify
+public bool multiSigVerify (const ref Signature signature,
+    const ref Point sumV, const ref Scalar c) nothrow @nogc @safe
+{
+    Sig sigRs = Sig.fromBlob(signature);
+    return sigRs.s.toPoint() == sigRs.R + (sumV * c);
+}
+
+/// multi-sig combine
+public Signature multiSigCombine (S)(const S signatures) nothrow @nogc @safe
+{
+    auto sigs = signatures.map!(x => Sig.fromBlob(x));
+    Point sum_R = sigs.map!(x => x.R).sum(Point.init);
```

Comment:
Are you sure this is correct? I would double-check this code, because the sum point of `A` and `B` should equal `A + B`, and not `Point.init + A + B`. I am pretty sure the two values are very different on the curve.

Reply:
The commit before this one updated addition of Points so that `Point.init + A = A`.

Reply:
Oh then that's really cool. But can you add a test? Or is it there?

Reply:
It is there in the same commit :-)